test/uitest.c

```diff
@@ -59,9 +59,11 @@ static int test_old()
     char pass[16];
     int ok = 0;
 
-    if ((ui_method =
-         UI_UTIL_wrap_read_pem_callback(test_pem_password_cb, 0)) == NULL
-        || (ui = UI_new_method(ui_method)) == NULL)
+    ui_method = UI_UTIL_wrap_read_pem_callback(test_pem_password_cb, 0);
+    if (!TEST_ptr(ui_method))
+        goto err;
+    ui = UI_new_method(ui_method);
+    if (!TEST_ptr(ui))
```

Comment:
Another possibility is this:

``` C
    if (!TEST_ptr(ui_method =
                 UI_UTIL_wrap_read_pem_callback(test_pem_password_cb, 0))
        || !TEST_ptr(ui = UI_new_method(ui_method)))
```

The benefit with that is that if there's an error printout, it will show the operation that failed and not just the variable.  I noticed that @pauli-oracle mastered this way of doing things in quite a number of his rewrites.